### PR TITLE
Update Istio addon to 0.6.0 and mirror images in gcr

### LINKED ISSUE #61910: Istio addon images should be in gcr.io for scanning
/kind bug

**What happened**:
https://github.com/kubernetes/kubernetes/pull/59378 added istio manifests to the kubernetes addons directory. There are several issues with the images in the yaml:

prom/statsd-exporter:v0.5.0
docker.io/prom/prometheus:v2.0.0
docker.io/openzipkin/zipkin:latest

If we want Istio to be considered a GKE Addon, these should all be mirrored to gcr.io so that we can continuously scan them for known vulnerabilities.
As a side note, the image should be fully specified (e.g. "docker.io/..." or "gcr.io/...").
The zipkin image should specify a version, so we don't get arbitrarily broken.

**What you expected to happen**:
Images should conform to guidelines above.

**How to reproduce it (as minimally and precisely as possible)**:
Inspect the yaml files. 

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v.1.10.0+

### PR DESCRIPTION
**Pull request by ostromart**
**What this PR does / why we need it:**
PR addresses two issues:
1. Update Istio addon version to latest stable 0.6.0 (required for e2e test integration).
2. Mirror images in gcr.io for security scanning, since this is an addon.

Fixes #61910 

**Release note**:
```release-note
Update version of Istio addon from 0.5.1 to 0.6.0.
See https://istio.io/about/notes/0.6.html for full Isto release notes.
```

### CODE REVIEW

File: cluster/addons/istio/auth/istio-auth.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ kind: Namespace
 metadata:
   name: istio-system
   labels:
+    istio-injection: enabled
```
Comment (yusuoh) [inline cluster/addons/istio/auth/istio-auth.yaml]: This should not be applied to istio-system namespace. It should be applied to the "default" namespace.

Comment (ostromart) [reply cluster/addons/istio/auth/istio-auth.yaml]: Done.

File: cluster/addons/istio/auth/istio-auth.yaml [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
 # TO UPDATE, modify files in install/kubernetes/templates and run install/updateVersion.sh
+#
+# Required for Istio as k8s addon.
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: default
+  labels:
+    istio-injection: enabled
+    addonmanager.kubernetes.io/mode: Reconcile
```
Comment (yusuoh) [inline cluster/addons/istio/auth/istio-auth.yaml]: Will this prevent users to add other labels to default namespace?

Comment (ostromart) [reply cluster/addons/istio/auth/istio-auth.yaml]: No, only items explicitly set in the yaml are reconciled (so you only cannot remove this label). Users are free to add anything not listed in the yaml.

File: cluster/addons/istio/auth/istio-auth.yaml [hunk 3]
```diff
@@ -593,6 +603,25 @@ spec:
...
+  labels:
+    k8s-app: istio
+    addonmanager.kubernetes.io/mode: Reconcile
+    package: solarwinds
+    istio: mixer-adapter
+spec:
+  group: config.istio.io
+  names:
+    kind: solarwinds
+    plural: solarwindses
```
Comment (cjcullen) [inline cluster/addons/istio/auth/istio-auth.yaml]: This is weird plural :).

**Comment by cjcullen** [comment]
The container image updates look fine to me, but this PR does a bunch of other stuff as well. I'd like somebody working on Istio to give this an LGTM too.

File: cluster/addons/istio/auth/istio-auth.yaml (hunk 3 above)
Comment (ostromart) [reply cluster/addons/istio/auth/istio-auth.yaml]: I think it's middle earth english - one hobbit, two hobbitses.

**Comment by ostromart** [comment]
@yusuoh works on Istio. Are you looking for someone in a specific function? 
btw. This yaml is a slightly modified version of https://github.com/istio/istio/releases/ 0.6. The differences are:
- k8s addon specific labels added for addon manager
- images come from gcr.io
- all istio extensions (prom, zipkin etc) are installed

**Comment by cjcullen** [comment]
I'm fine adding the official LGTM if @yusuoh says it looks good.

**Comment by ostromart** [comment]
could you please go ahead and lgtm the merge @cjcullen ?

**Comment by cjcullen** [comment]
@zmerlynn can you approve?

**Comment by krzyzacy** [comment]
/assign @mikedanese 
@zmerlynn is OOO, Mike can you give an approval here?

**Comment by krzyzacy** [comment]
(tests passed)

**Pull request merged**